From c563db5aff38cd840806f2788731ba3dfe547c7f Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Tue, 3 Dec 2024 15:10:31 +0800
Subject: [PATCH] fix-std-filesystem-error

---
 src/lib/OpenEXR/ImfStdIO.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/src/lib/OpenEXR/ImfStdIO.cpp b/src/lib/OpenEXR/ImfStdIO.cpp
index 9d86850..3783cca 100644
--- a/src/lib/OpenEXR/ImfStdIO.cpp
+++ b/src/lib/OpenEXR/ImfStdIO.cpp
@@ -14,7 +14,6 @@
 #include <ImfMisc.h>
 #include <ImfStdIO.h>
 #include <errno.h>
-#include <filesystem>
 
 using namespace std;
 #include "ImfNamespace.h"
@@ -27,14 +26,14 @@ namespace
 inline ifstream*
 make_ifstream (const char* filename)
 {
-    return new ifstream (std::filesystem::path (filename),
+    return new ifstream (filename,
                          ios_base::in | ios_base::binary);
 }
 
 inline ofstream*
 make_ofstream (const char* filename)
 {
-    return new ofstream (std::filesystem::path (filename),
+    return new ofstream (filename,
                          ios_base::out | ios_base::binary);
 }
 
-- 
2.20.1

