From ff5d25e1f669b90aa3679595d21b3f0faeece416 Mon Sep 17 00:00:00 2001
From: Your Name <you@example.com>
Date: Sun, 13 Jul 2025 10:14:21 +0800
Subject: [PATCH] fix-cxxversion

---
 cmake/CMakeLists.txt | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/cmake/CMakeLists.txt b/cmake/CMakeLists.txt
index 61cb3eb..e4fefb4 100644
--- a/cmake/CMakeLists.txt
+++ b/cmake/CMakeLists.txt
@@ -258,10 +258,6 @@ option(onnxruntime_USE_VITISAI_INTERFACE "Build ONNXRuntime shared lib which is
 option(onnxruntime_USE_QNN_INTERFACE "Build ONNXRuntime shared lib which is compatible with QNN EP interface" OFF)
 
 
-if("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" AND CMAKE_C_COMPILER_VERSION VERSION_LESS 11.1)
-  message(FATAL_ERROR  "GCC version must be greater than or equal to 11.1")
-endif()
-
 # ENABLE_TRAINING includes all training functionality
 # The following 2 entry points
 # 1. ORTModule
-- 
2.25.1

