diff --git a/CMakeLists.txt b/CMakeLists.txt
index c32c332..26d1f28 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,5 +16,6 @@ include(AppendCompilerFlags)
 
 project(FastPFOR CXX C)
+include(GNUInstallDirs)
 set(PROJECT_URL "https://github.com/fast-pack/FastPFOR")
 set(PROJECT_DESCRIPTION "The FastPFOR C++ library: Fast integer compression")
 # Need to bump this when we release a new version.
