--- a/ssl/ssl_lib.c
+++ b/ssl/ssl_lib.c
@@ -2380,7 +2380,8 @@ int SSL_export_keying_material(SSL *s, unsigned char *out, size_t olen,
                                const unsigned char *p, size_t plen,
                                int use_context)
 {
-    if (s->version < TLS1_VERSION && s->version != DTLS1_BAD_VER)
+    if (s->version < TLS1_VERSION && s->version != DTLS1_BAD_VER
+        && s->version != GMTLS_VERSION)
         return -1;
 
     return s->method->ssl3_enc->export_keying_material(s, out, olen, label,
