--- a/crypto/sm2/sm2_exch.c
+++ b/crypto/sm2/sm2_exch.c
@@ -579,9 +579,18 @@
 int SM2_compute_share_key(unsigned char *out, size_t *outlen,
 	const EC_POINT *peer_ephem, EC_KEY *ephem,
 	const EC_POINT *peer_pk, const unsigned char *peer_z, size_t peer_zlen,
 	const unsigned char *z, size_t zlen, EC_KEY *sk, int initiator)
 {
-    
-	memset(out, 1, *outlen);
-	return 1;
+	(void)out;
+	(void)outlen;
+	(void)peer_ephem;
+	(void)ephem;
+	(void)peer_pk;
+	(void)peer_z;
+	(void)peer_zlen;
+	(void)z;
+	(void)zlen;
+	(void)sk;
+	(void)initiator;
+	return 0;
 }
